app/views/admin/api_docs/base/swagger.html.erb

```diff
@@ -99,16 +161,23 @@
 
   <script type="text/javascript">
     $(function () {
-      var url = "<%= preview_admin_api_docs_service_path(@api_docs_service, format: :json)-%>";
+      var host = "<%= @api_docs_service.host_with_port %>";
+      var spec = <%= ActionController::Base.helpers.sanitize(@api_docs_service.body) %>;
+
+      injectAuthParams(spec);
+
       window.swaggerUi = new SwaggerUi({
-        url: url,
+        spec: spec,
```

Comment:
Is it possible to refactor `injectAuthParams` so, instead of changing the state, it actually returns a new `spec` and then we replace this with `spec: injectAuthParams(spec)`?

Reply:
Yes it is, but is there any particular reason?

Reply:
It's always better when you know your function does not change the state IMO and in this case it's not needed to anyway.

Reply:
The problem is that this is all happening in the browser, so we are not only filling users's memory with the whole JSON once but we want to do a deep copy of it in order not to modify it.

You definitely have a point here, I just don't think it is worth it in this particular case.

We could actually argue we are even changing the app's state by doing this. We are retrieving an object, we are working with it and finally using it. There is no state involved.